eclipse/openvsx
# Store and send missing fields in extension metadata

Fixes #330

### Testing steps
- Run in Gitpod, wait until the test extensions are published
- Using `psql`, query:
```
SELECT name, published_date, last_updated_date FROM extension;
```
The `published_date` and `last_updated_date` columns should have timestamp values.
- Call the `/extensionquery` endpoint:
```
curl --location --request POST 'http://localhost:8080/vscode/gallery/extensionquery' \
--header 'Content-Type: application/json' \
--data-raw '{"filters":[{"criteria":[{"filterType":7,"value":"redhat.vscode-yaml"},{"filterType":8,"value":"Microsoft.VisualStudio.Code"},{"filterType":12,"value":"4096"}],"pageNumber":1,"pageSize":1,"sortBy":0,"sortOrder":0}],"assetTypes":[],"flags":950}'
```
The response should contain `releaseDate`, `publishedDate` and `lastUpdated` fields
- Publish another version of `redhat.vscode-yaml`. The  `lastUpdated` / `last_updated_date` should be updated to current timestamp.
- To test the migration you can use `psql` to revert the migration:
```
DELETE FROM flyway_schema_history WHERE version = '1.22';
ALTER TABLE extension DROP COLUMN published_date;
ALTER TABLE extension DROP COLUMN last_updated_date;
```
Restart the server. Once the server is running you can check that `published_date` and `last_updated_date` columns are added and have timestamp values, using `psql`:
```
SELECT name, published_date, last_updated_date FROM extension;
```
